pkg/provider/azure_standard.go

```diff
@@ -526,25 +526,31 @@ func (as *availabilitySet) GetInstanceIDByNodeName(name string) (string, error)
 }
 
 // GetPowerStatusByNodeName returns the power state of the specified node.
-func (as *availabilitySet) GetPowerStatusByNodeName(name string) (powerState string, err error) {
+func (as *availabilitySet) GetPowerStatusByNodeName(name string) (powerState string, provisioningState string, err error) {
 	vm, err := as.getVirtualMachine(types.NodeName(name), azcache.CacheReadTypeDefault)
 	if err != nil {
-		return powerState, err
+		return powerState, provisioningState, err
 	}
 
+	if vm.VirtualMachineProperties == nil || vm.VirtualMachineProperties.ProvisioningState == nil {
```

Comment:
I'd prefer if we return a the nil status to the caller rather than default to something. This code is only called by `InstanceShutdownByProviderID` which is called by the cloudprovider's by "shutdownInCloudProvider".

I don't see it being an issue if we just return false instead of assuming it's stopped. Why would we want to taint a node that is under deletion with a shutdown annotation prematurely? 

It should already get the unreachable/non-executable taint

Reply:
confirmed ProvisioningState would never be empty, would change the logs message